src/features/calendarFilter/ui/DateSlider.tsx

```diff
@@ -1,17 +1,91 @@
-import {CalendarCard} from "@/shared/ui/CalendarCard";
-import {SlickSlider} from "@/shared/ui/SlickSlider/SlickSlider";
-import {ICardProps } from "@/shared/types";
-import {generateCalendarDays} from "../lib/generateCalendarDays";
-import {currentMonth, settings} from "../model/constants";
+import { CalendarCard } from "@/shared/ui/CalendarCard";
+import { SlickSlider } from "@/shared/ui/SlickSlider/SlickSlider";
+import { ICardProps } from "@/shared/types";
+import { generateCalendarDays } from "../lib/generateCalendarDays";
+import { currentMonth, settings, generateDateFormat } from "../model/constants";
+import { useAppSelector, useAppDispatch } from "@/shared/model";
+import {
+  startDateSetted,
+  endDateSetted,
+  startDateGTESetted,
+} from "@/features/searchFilter/model/SearchFilterSlice";
+import Svg from "@/shared/ui/Svg";
 
 export function DateSlider() {
   const dateArr = generateCalendarDays(40);
-  const cards = dateArr.map((el: ICardProps, index: number) => <CalendarCard key={index} date={el.date} weekDay={el.weekDay} />);
+  const dispatch = useAppDispatch();
+  const { startDate, endDate, startDateGTE } = useAppSelector(
+    (state) => state.searchFilter
+  );
+
+  const onHandleCloseCalendarFilter = () => {
+    dispatch(startDateSetted(""));
+    dispatch(endDateSetted(""));
+    dispatch(startDateGTESetted(""));
+  };
+
+  const onHandleClickDate = (date: ICardProps) => {
+    if (startDate === "") {
```

Comment:
you can change all contructions like (startDate === "") with (startDate), or (startDate != "") with (!startDate)